### PR TITLE
3062 list style audit

### LINKED ISSUE #3062: [List] Style audit
### What package is this for?

- [x] React
- [ ] Angular

## 1. Header

The padding to the left and the right of the header should be 16px — it is currently at 12px

### Current

Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--basic-single-column

Screen Shot 2021-11-15 at 2 14 52 PM

Screen Shot 2021-11-15 at 4 20 16 PM

## 2. Search

Update search space to be a search container that fills the full row. The row height should be the same as the row items (40px)

### Current

Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--basic-single-column-with-search

Screen Shot 2021-11-15 at 2 25 33 PM

Screen Shot 2021-11-15 at 2 28 55 PM

### Proposed

Screen Shot 2021-11-15 at 2 45 43 PM

Screen Shot 2021-11-15 at 2 45 11 PM

## 3. Overflow icon 

Fix overflow icon button size to match single icon size and placement

### Current
Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--with-row-actions-multiple

Screen Shot 2021-11-15 at 2 50 35 PM

Placement of the overflow menu looks detached as well

Screen Shot 2021-11-15 at 2 50 46 PM

### Proposed

Match icon button placement and size to this storybook item: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--with-row-actions-single
Screen Shot 2021-11-15 at 2 51 21 PM
Screen Shot 2021-11-15 at 2 54 29 PM

## 4. Empty state

Update the empty state to use the newer empty state pattern (left aligned)

### Current

Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--with-empty-state

Screen Shot 2021-11-15 at 2 56 38 PM

### Proposed

Empty state storybook existing component: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-emptystate--first-time-use

Screen Shot 2021-11-15 at 3 16 12 PM

## 5. Pagination footer

The text in the footer should be 16px from the left to line up with the content. There also shouldn't be a border around the full component, as it creates a double border at the top and unnecessary border around the sides and bottom. 

### Current

Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--with-pagination

Screen Shot 2021-11-15 at 3 19 18 PM

### Proposed

PAL guidance: https://pages.github.ibm.com/ai-applications/design/components/pagination/overview#list-pagination 
(this came from the former IoT PAL but is still correct for the list component footer)

Screen Shot 2021-11-15 at 3 24 48 PM

## 6. Load more

This seems like it was a request that came from a team but wasn't approved by PAL. 
Generally, any loading state should use the skeleton state (which is already available in the knobs as isLoading).

A better experience would be a lazy load where the rows loading automatically happened when reaching the end of the current populated list, but I'm not sure if that's available/a team technical limitation.If there needs to be a manual load trigger, the text shouldn't be center aligned, but left aligned to line up with the row text. In this case, if there is a hierarchy, then it should line up with the level it is associated with. 

### Current

Storybook: 

Placement
Screen Shot 2021-11-15 at 3 26 32 PM

Loading state
Screen Shot 2021-11-15 at 3 27 36 PM

### Proposed

Placement - can we also update the text to say "View more" instead of "Load more..."

Screen Shot 2021-11-15 at 4 04 10 PM

Loading state (skeleton state)
Screen Shot 2021-11-15 at 3 40 21 PM

## 7. Mixed hierarchies

In the mixed hierarchy (which is actually called hierarchy), the parent levels and nested levels that expand should not be semi-bold in text. All levels should be using `body-short-01` and `text-02` color. Only in the regular hierarchy list (actually called categories) should the parent levels be semi-bold/productive-heading-01.

### Current
Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-hierarchylist--with-mixed-hierarchies

Screen Shot 2021-11-15 at 4 08 47 PM

### Proposed

Screen Shot 2021-11-15 at 4 11 27 PM

## 8. Tall row

Adjust the height and text alignment in the large row

### Current

Storybook: https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-list--with-is-large-row-and-icon

Screen Shot 2021-11-15 at 4 13 26 PM

Screen Shot 2021-11-15 at 4 19 37 PM

### Proposed
Screen Shot 2021-11-15 at 4 19 10 PM

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #3062

**Summary**
Fixes and improves a lot of design issues for the List and the components using the List

**Change List (commits, features, bugs, etc)**
See #3062 for specification and conversation. The code is best view per major commit:

1 - 2. Header and Search: Fix spacing and have search fill full row https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/0d62347e3f9ed3b253e773501bc4ebc8d68fbc03
3. Overflow icon: Fix overflow icon button size to match single icon size and placement. A previous fix to the button had to be adjusted to allow small ghost icon buttons. Updated stories that weren't using real buttons. https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/66bd1b278e43b04ce73c04d6fc610cb77132f710
4. Empty state. Add empty state for empty list and empty search result https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/008a23b60430020ce56018f058d674b8d0d5eb96
5. Pagination footer: Fix height, border and text alignments. https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/0240d21786c3749e6ed70d9fae681422fbe9192a
6. Load more: Align with level content, change loader and label. https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/460855f4ad0e4e7f72b441d5be7f52068265e32d
7. Mixed hierarchies: Fix indentation so that every child is always indented. Removed isCategory from Story to match design. https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/e25b8acc188df63ecee7b7eaa089ee46d652ce64
8. Tall row: Make tall rows (using isLarge) shorter. This required a redesign of the ListItem layout. Instead of absolute positioned elements corrected by special padding that had become very brittle and difficult to modify (and also failed in a few scenarios e.g. when large/tall rows with actions and labels) we now have flexbox with column gap. https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3141/commits/36c9a1048b10e959b5a5cdfa1db0a640ca530512 This also fixed #3136.

**Acceptance Test (how to verify the PR)**
Note! Stories using document.dir for `tooltipPosition` or `flipped` props are not picking up the changes from the RTL knob. To properly test change the RTL knob, leave the story and enter again to see it working. 

_1. Header_
Verify that the headers matches the spec (aligned with content)
1. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--basic-single-column 
2. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-%E2%98%A2%EF%B8%8F-tablecolumncustomizationmodal--playground-with-knobs

_2 Search_
Verify that the search matches the spec (full width when selected and icon aligned with content)
1. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--basic-single-column-with-search
2. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-selectusersmodal--default-modal

_3 Overflow icon_
Verify that the actions using the menu overflow matches the spec, i.e. the the small buttons are used. Also check that the single row action buttons are aligned with the new spec. 

1. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-listitem--with-single-row-action
2. Make sure the action button size is according to spec 
3. Enable RTL and make sure the button has not grown wider than in LTR
4. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-listitem--with-single-row-action 
5. Make sure the overflow action button size is according to spec and that the menu is correctly positioned
6. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--list-with-multiple-actions and make sure action buttons look ok.
7. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-overflowmenu--basic and check out the new size knob to make sure the `sm` size is being rendered ok.

_4 Empty state_
Verify that the empty states matches the spec
1. Verify Empty list  https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-empty-state
2. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--basic-single-column. 
3. Enable knob "use search"  and type `xyz` in the search field
4. Verify that the "No results found" with a magnifier is used
5. Verify that the knobs "Empty state text" (combine with knob "demo empty list") and "Empty search state text" can modify the messages
6. Verify that the "No results found" is shown when search returns empty in story https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--stateful-list-with-nested-searching
7. Verify the custom empty state in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-empty-state
8. Verify empty list and empty search in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-selectusersmodal--default-modal
9. Verify custom empty states in story https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-%E2%98%A2%EF%B8%8F-tablecolumncustomizationmodal--with-many-columns-and-load-more
10. Verify empty search result state in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-tablemanageviewsmodal--with-callbacks-implemented (this search is external to the list and therefore looks different)
11. Verify the empty search state in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--basic

_5 Pagination footer_
Verify that pagination is according to te spec. Verify that the text "Page x" is correctly aligned to the left and has the correct height and no border on the sides nor bottom.
1. Verify that it looks ok with knob "isFullHeight" true and false. Double scrolling is  https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-pagination
2. Repeat for story https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--stateful-list-with-nested-searching. Also expand and collapse categories to verify the pagination.
3. Same for this story https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--list-with-empty-row

_6 Load more_
Verify that the load more follows the spec. Aligns with content per level. Uses skeleton loader.
1. Go to story https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-load-more
2. Verify that the text is "View more" and that it is aligned on the first level
3. Expand "Site 3" and verify that the  "View more" is aligned with thet content on that level
4. Click on a "View more" and verify that we get the skeleton loader
5. Check it all again with `isVirtualList` knob active

_7 Mixed hierarchies_
Verify that the stories are up to date with the spec. Not explicitly stated in the spec text but shown in the images are that all children should have indentation in relation to their parent.
1. Check that there all [List stories](https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--basic-single-column) and [HierarchyList stories](https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--stateful-list-with-nested-searching) with categories (bold texts) only have one level of nesting. 
2. Verify that all children have indentation in relation to their parent in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-hierarchy 
3. Check indentation in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-checkbox-multi-selection-and-hierarchy
4. Check indentation in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-nested-reorder
5. and here https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-mixed-hierarchies
6. and here https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-%E2%98%A2%EF%B8%8F-tablecolumncustomizationmodal--with-column-groups 

_8 Tall row_ 
Verify that the tall row (enabled by prop `isLargeRow`) is always 64 px tall. It must work with and without secondary values, with extremely long values, with single and multiple row actions, with checkboxes, with icons, tags, hierarchies and categories, with isVirtual false and true . These things should all be properly aligned and spaced. They should look and work ok separately and combined. 

1. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-checkbox-multi-selection-and-hierarchy Enable isLargeRow knob and test with the different knobs available. 
2. Enable isLargeRow knob in https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-tags
3. Verify https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-is-large-row-and-icon
4. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-listitem--with-is-large-row
5. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-listitem--testing-is-large-row-overflow
6. Verify with the all knobs here https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-listitem--basic-w-knobs
7. Go to https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-%E2%98%A2%EF%B8%8F-tablecolumncustomizationmodal--playground-with-knobs and enable knob "Column key used for secondary value (secondaryValue)"
8. Verify with all the knobs for that story 
9. Most of the [stories for the HierarchyList](https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--stateful-list-with-nested-searching) have an isLargeRow knob. Verify that they look ok
10. Verify https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--list-with-large-row
11. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--large-row-list-with-multiple-actions
12. https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--large-row-list-with-overflow-menu

**Regression Test (how to make sure this PR doesn't break old functionality)**
No changes other than the improvements described by this PR in

- All List stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-list--basic-single-column
- All HierarchyList stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-hierarchylist--stateful-list-with-nested-searching
- All ListCard stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-listcard--basic
- All ListItem stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-listitem--basic-w-knobs
- All SimpleList stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-simplelist--basic
- All TableColumnCustomizationModal stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table-%E2%98%A2%EF%B8%8F-tablecolumncustomizationmodal--playground-with-knobs
- This TableManageViewsModal https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-tablemanageviewsmodal--with-callbacks-implemented
- All ListBuilder stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-listbuilder--complex-nested-example
- All SelectUsersModal stories https://deploy-preview-XXXXX--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-selectusersmodal--default-modal
- All Button stories look the same when testing different knob configurations (size, type, icon etc)

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by JordanWSmith15** [comment]
I would like if @dianatran18 would do a review of this PR

File: packages/react/src/components/Header/_header.scss
Comment (bjornalm) [inline packages/react/src/components/Header/_header.scss:69]: Moved this one out of the global scope

**Comment by dianatran18** [comment]
I'm going to follow up with a small change on the list headers once I confirm something with a few teams, but one thing I noticed was that there's an inconsistency with the selected state of a parent row for the checkbox multi-selection and hierarchy use case. 

To reproduce:
- Go to [List: with checkbox multi-selection and hierarchy](https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-checkbox-multi-selection-and-hierarchy)
- Expand one parent row and select a child to create indeterminate parent state
- Next, select a different parent row, then expand the parent row, and deselect a child to create indeterminate parent state
- The first parent row will not have a selected background and blue highlight, but the second parent row will

See row 2 and 3 as the example
Screen Shot 2021-12-13 at 3 16 04 PM

I think we can have it follow the 3rd row in the example, and show the selected state of the row

**Review by kevinsperrine — CHANGES_REQUESTED** [review]
- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--basic-single-column-with-search
Turn on RTL. Should the icon be placed on the right?
Screen Shot 2021-12-13 at 4 22 25 PM

<br/>
<br/>

- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-selectusersmodal--default-modal
Use left-pointing arrow in RTL?
Screen Shot 2021-12-13 at 4 24 13 PM

<br/>
<br/>

- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--list-with-multiple-actions
Should we open the tooltips top instead of left/right?
list-actions-tooltip

<br/>
<br/>

- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-simplelist--list-with-multiple-actions
In RTL, The header buttons have margins on the left making too much space in RTL
Screen Shot 2021-12-13 at 5 10 43 PM
```css
.iot--list-header--btn-container > * {
    margin-left: 1rem;
}
```

<br/>
<br/>

- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-load-more
when turning on "isVirtualList" the "View more" text doesn't align the same as when isVirtualList is false.
Screen Shot 2021-12-13 at 4 47 04 PM

<br/>
<br/>

- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-%E2%98%A2%EF%B8%8F-tablecolumncustomizationmodal--playground-with-knobs
~When dragging rows the blue line doesn't extend all of the way to the end of the list item~ Edit: actually, on second thought, this is probably intended because the padding is now applied to the list container, right?
Screen Shot 2021-12-13 at 4 59 19 PM

<br/>
<br/>

- [ ] https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-mixed-hierarchies
The "Task 1" and "My Reports" text seems mis-aligned, no?
Screen Shot 2021-12-13 at 5 06 32 PM

<br/>
<br/>

I _think_ that's everything, but I did not go through every story in the regression test list. I'll check that again tomorrow.

File: packages/react/src/components/OverflowMenu/_overflow-menu.scss
Comment (kevinsperrine) [reply packages/react/src/components/OverflowMenu/_overflow-menu.scss:11]: In https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-statefultable--stateful-example-with-expansion-max-pages-and-column-resize the overflow menu "spacer" is too large, but I _think_ that's because my PR where size is passed from the table down to the RowActionsCell hasn't been merged into this PR, right? 

Screen Shot 2021-12-13 at 5 26 13 PM

File: packages/react/src/components/Table/TableColumnCustomizationModal/TableColumnCustomizationModal.test.jsx
Comment (bjornalm) [reply packages/react/src/components/Table/TableColumnCustomizationModal/TableColumnCustomizationModal.test.jsx:601]: Because the secondary value now appears inside the `...values--main` container

File: packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap
Comment (bjornalm) [reply packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:23828]: @davidicus what is your take on changing a default i18n string?

**Comment by bjornalm** [comment]
> ...
> * Go to [List: with checkbox multi-selection and hierarchy](https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-checkbox-multi-selection-and-hierarchy)
> * Expand one parent row and select a child to create indeterminate parent state
> * Next, select a different parent row, then expand the parent row, and deselect a child to create indeterminate parent state
> * The first parent row will not have a selected background and blue highlight, but the second parent row will
> 
> I think we can have it follow the 3rd row in the example, and show the selected state of the row

I've created a new issue for this since it is not new for this PR and probably just an issue of List **story** and not the list itself. See more info in https://github.com/carbon-design-system/carbon-addons-iot-react/issues/3148

File: packages/react/src/components/Header/_header.scss
Comment (bjornalm) [reply packages/react/src/components/Header/_header.scss:71]: Will do :-)

File: packages/react/src/components/List/ListContent/ListContent.jsx
Comment (bjornalm) [reply packages/react/src/components/List/ListContent/ListContent.jsx:256]: Thanks :-)

File: packages/react/src/components/List/ListItem/_list-item.scss
Comment (bjornalm) [reply packages/react/src/components/List/ListItem/_list-item.scss:185]: Not needed. Thanks

File: packages/react/src/components/List/ListItem/_list-item.scss
Comment (bjornalm) [reply packages/react/src/components/List/ListItem/_list-item.scss:232]: Good catch

File: packages/react/src/components/List/VirtualListContent/VirtualListContent.jsx
Comment (bjornalm) [reply packages/react/src/components/List/VirtualListContent/VirtualListContent.jsx:163]: I wanna say no, but the right answer is yes :-) I'll fix it

File: packages/react/src/components/OverflowMenu/_overflow-menu.scss
Comment (bjornalm) [reply packages/react/src/components/OverflowMenu/_overflow-menu.scss:11]: Yes, that issue was created to prevent this, and now that it is merged the problem is solved.

File: packages/react/src/components/List/List.story.jsx
Comment (bjornalm) [reply packages/react/src/components/List/List.story.jsx:325]: Yes, I thought I had found them all

File: packages/react/src/components/List/ListContent/ListContent.jsx
Comment (bjornalm) [reply packages/react/src/components/List/ListContent/ListContent.jsx:89]: Yea, it would never actually be used (which makes this a point for our tech spec discussion) but we should be consistent here. I'll change it.

**Comment by davidicus** [comment]
Getting a double scrollbar when `isFullHeight` is false. 

>5 Pagination footer
Verify that pagination is according to te spec. Verify that the text "Page x" is correctly aligned to the left and has the correct height and no border on the sides nor bottom.

>Verify that it looks ok with knob "isFullHeight" true and false. Double scrolling is https://deploy-preview-3141--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-list--with-pagination

image

**Comment by bjornalm** [comment]
> Getting a double scrollbar when `isFullHeight` is false.

This is an old bug with the list, but I'll add a fix to this PR

**Comment by dianatran18** [comment]
One additional thing to check on is the header styling. This is the [current style](https://pages.github.ibm.com/ai-applications/design/components/list/style#list-header) we have defined in our list spec, which was adjusted to match the data table header styling more. 

Current:
Screen Shot 2022-01-04 at 11 23 35 AM

Design:
image

I'm trying to back track to see if/when we got the request for batch actions in a list from a team, but in general the main difference is the height is 48px vs 56px (current), and the primary button is full bleed. 

Note: one thing we'll need to adjust in the design spec is that the header should be using `$productive-heading-02` as it is in the current implementation. 

——— 

Examples of how teams are using lists now, which looks like the 48px height should be okay.

Graphite:
image

ELM: 
Screen Shot 2022-01-04 at 11 22 20 AM

File: packages/react/src/components/List/ListHeader/_list-header.scss
Comment (kevinsperrine) [inline packages/react/src/components/List/ListHeader/_list-header.scss:23]:
```suggestion
    margin-right: -($spacing-03);
```

**Review by kevinsperrine — APPROVED** [review]
Just some parens on a negative variable. Otherwise, the code looks great to me. I'll check stories now.

**Review by davidicus — APPROVED** [review]
LGTM once Diana's comments are addressed.

**Comment by dianatran18** [comment]
I'll move my last comment out to a separate issue to not block this PR!

**Pull request merged**
